39 create contributor guidelines

emily-bicks opened the pull request:
Closes #39 adds contributor guidelines to readme. Closes #34

Commit:
adding contributor guidelines to readme

Commit:
fixing code blocks in readme

Commit:
fixing bold formatting

Commit:
fixing newline in readme

Commit:
moving to separate contributor guidelines file

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+8. Add new functions to corresponding scripts, or new scripts if feature doesn't fit within existing scripts
+9. Add tests for any new functions to corresponding test script located within tests folder
+10. Update documentation within nesta_ds_utils>docs>source
+
```
```suggestion
```

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+
+To add a feature:
+
+1. Add an issue that describes the feature
```
I think we should encourage people to start a conversation (here or on Slack?) about the issue/feature before going ahead and developing something. It might already exist, or there might be a reason it doesn't, or it might non-obviously be a bad idea.

emily-bicks replied:
Yeah - maybe these guidelines should start from after a feature has already been proposed and approved for working on. I will create separate instructions for suggesting a feature.

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+
+All contributions to this project should follow the [Nesta Github Guidelines](https://github.com/nestauk/github_support/blob/dev/guidelines/README.md)
+
+To add a feature:
```
Does this mean there should be separate instructions for e.g. reporting a bug?

emily-bicks replied:
Added separate instructions for reporting bug/requesting feature and made these for general development

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+10. Update documentation within nesta_ds_utils>docs>source
+
+    - If just adding a new feature to an existing script modify the .rst files corresponding to the script and tests that you modified
+    - If adding a new script, add new .rst files for your script and test script, modify index.rst to include your scripts, and modify conf.py to import your scripts
+
+      **all of your functions must have properly formatted docstrings (we are using [Google](https://sphinxcontrib-napoleon.readthedocs.io/en/latest/example_google.html) format) to be picked up by the documentation**
+
+11. Open a PR and commit your changes to the branch you created
```
I think these broken lines are messing up the list formatting a bit.

emily-bicks replied:
I purposefully did that to indent the docs instructions under the bullet about adding to the docs, but can change it if you think it looks weird

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+1. Add an issue that describes the feature
+2. Clone the repository using SSH protocol
+3. Create a conda environment with python >= 3.8
+4. cd into the top level nesta_ds_utils directory
```
```suggestion
4. cd into the top level `nesta_ds_utils` directory
```

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+7. Create a new branch corresponding to the issue: `git checkout -b [ISSUE NUMBER]_[BRIEF DESCRIPTION]` (ex: 10_fix_docstrings)
+8. Add new functions to corresponding scripts, or new scripts if feature doesn't fit within existing scripts
+9. Add tests for any new functions to corresponding test script located within tests folder
+10. Update documentation within nesta_ds_utils>docs>source
```
Should this be `nesta_ds_utils/docs/source`?

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+5. Run `pip install -e ."[dev]"` to install the package with the developer requirements
+6. Run `pre-commit install` to setup pre-commit
+7. Create a new branch corresponding to the issue: `git checkout -b [ISSUE NUMBER]_[BRIEF DESCRIPTION]` (ex: 10_fix_docstrings)
+8. Add new functions to corresponding scripts, or new scripts if feature doesn't fit within existing scripts
```
Replace "scripts" with "modules"?

georgerichardson commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,24 @@
+To add a feature:
+
+1. Add an issue that describes the feature
+2. Clone the repository using SSH protocol
```
```suggestion
2. Clone the repository (we typically use the SSH protocol)
```

georgerichardson reviewed (commented):
This looks great overall. Have made a few specific comments.

My only other suggestion is to also make a template issue on GitHub for people to raise feature requests with some basic headers (maybe: Description, Examples and Links to existing code?). Then if possible, link to that issue template straight from CONTRIBUTING.md

Commit:
adding bug report template

Commit:
adding feature request template

emily-bicks commented:
I have added two issue templates here: https://github.com/nestauk/nesta_ds_utils/tree/39_create_contributor_guidelines/.github/ISSUE_TEMPLATES

I have also added descriptions for submitting bugs/feature requests to the contributor guidelines separate from actually developing.

Commit:
fixing formatting in contributor guidelines

Commit:
adding hyperlinks to slack

EnricoGavagnin commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,54 @@
+
+1. Post in the [#dev channel](https://nesta.slack.com/app_redirect?channel=dev) with the Issue number you plan to address
+2. Clone the repository (we typically use the SSH protocol)
+3. Create a conda environment with python >= 3.8
```
```suggestion
3. Create and activate a conda environment with python >= 3.8
```

EnricoGavagnin commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,54 @@
+10. Update documentation within `nesta_ds_utils/docs/source`:
+
+    - If just adding a new feature to an existing module modify the .rst files corresponding to the module and tests that you modified
+    - If adding a new module, add new .rst files for your module and test module, modify index.rst to include your modules, and modify conf.py to import your modules
```
Very minor, but maybe I would add the location of the .rst files. Maybe even linking to an example, so people unfamiliar with Sphinx might have a reference.

EnricoGavagnin commented on CONTRIBUTING.md:
```diff
@@ -0,0 +1,54 @@
+6. Run `pre-commit install` to setup pre-commit
+7. Create a new branch corresponding to the issue: `git checkout -b [ISSUE NUMBER]_[BRIEF DESCRIPTION]` (ex: 10_fix_docstrings)
+8. Add new functions to corresponding modules, or new modules if feature doesn't fit within existing modules
+9. Add tests for any new functions to corresponding test module located within tests folder
```
Shall we add a bullet point about adding dependencies to `setup.cfg`? Something like: 

- ?. Add any new package requirements in `setup.cfg` under `install_requires`. Requirements which are used only for testing should go under `dev` in `[options.extras_require]`.

EnricoGavagnin reviewed (approved):
It looks very good and I am sure this will be very useful to have people contributing! 

I left a couple of comments on the guidelines. I also reviewed the text of the issue templates, but I wasn't sure if there was a way to actually test these.

emily-bicks committed:
41 data manipulation (#44)

* adding basic date functions and tests

* adding data_manipulation functions to docs

* adding numpy to install requirements

* removing underscore in format

* allowing users to specify null values or list of possible date options

* adding tests for new datetime functionality

* adding additional date part options

* fixing documentation based on changes

* fixing formatting of error messages

Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.home>
Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.local>

emily-bicks committed:
Update CONTRIBUTING.md

Co-authored-by: George Richardson <georgerichardson@posteo.net>

emily-bicks committed:
Update CONTRIBUTING.md

Co-authored-by: Enrico Gavagnin <53174613+EnricoGavagnin@users.noreply.github.com>

Commit:
addressing comments

Commit:
tweaking guidelines

EnricoGavagnin committed:
23 saving figure altair (#43)

* precommit fixed - added plotting.py module with saving utils

* Added uni tests for saving plots with altair - precommit   xed

* Added .rst files for plotting and test_plotting - precommits fixed

* Added get driver function

* Added headless option chrome driver

* Added headless option chrome driver - precommits fixed

* deleted new lines, added version requires and other minor corrections - precommit fixed

* Remove _create_paths from plotting - precommit fixed

* Allow str type as filetypes in plotting.save- precommit fixed

* Change `plotting.save` to ccept bool variable, adjusted tests and other minor tweaks - precommits fixed

* 41 data manipulation (#44)

* adding basic date functions and tests

* adding data_manipulation functions to docs

* adding numpy to install requirements

* removing underscore in format

* allowing users to specify null values or list of possible date options

* adding tests for new datetime functionality

* adding additional date part options

* fixing documentation based on changes

* fixing formatting of error messages

Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.home>
Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.local>

* Added scale_factor as parameter in plotting.save - precommits fixed

* Consistent import in tests modules - precommits

* Added annotation for altair charts - precommits

* Changed .rst file for test_plotting

* precommit fixed - added plotting.py module with saving utils

* Added uni tests for saving plots with altair - precommit   xed

* Added .rst files for plotting and test_plotting - precommits fixed

* Added get driver function

* Added headless option chrome driver

* Added headless option chrome driver - precommits fixed

* deleted new lines, added version requires and other minor corrections - precommit fixed

* Remove _create_paths from plotting - precommit fixed

* Allow str type as filetypes in plotting.save- precommit fixed

* Change `plotting.save` to ccept bool variable, adjusted tests and other minor tweaks - precommits fixed

* Added scale_factor as parameter in plotting.save - precommits fixed

* Consistent import in tests modules - precommits

* Added annotation for altair charts - precommits

* Changed .rst file for test_plotting

* Ammend conf.py for plotting utils

Co-authored-by: emily-bicks <113300626+emily-bicks@users.noreply.github.com>
Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.home>
Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.local>

EnricoGavagnin committed:
24_styling_figures (#45)

* added nesta_theme and setup_theme in plotting - precommits

* added styling utils to docs

* added altair to requirements

* added Averta find function and warning

* added unit_test for styling figures-precommits

* added test_plotting.test_nesta_theme_activation to docs

* added type for return obj in nesta_theme()

* Load nesta_theme as yaml file - precommits

* Added pyyaml to requirements-preccommits

* basic changes to plotting utils. made nesta_theme into _load_nesta_theme and made setup_theme a bit more general in case we envision adding additional theme options

* removing nesta_theme from docs as it is private

Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.home>

emily-bicks committed:
adding action to build docs (#47)

* adding action to build docs

* fixing test docs for plotting functions

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* adding git statements to doc build

* remove token use test

* remove token use test

* simplifying theme

* build action

* update to use github token

* fix empty line

* fix indenting

* fix more indenting

* auto updating docs

Co-authored-by: Emily Bicks <emilybicks@NESTA-CQ6PWN4772.local>
Co-authored-by: Jack-Vines <jack.vines@live.co.uk>
Co-authored-by: github-actions[bot] <github-actions[bot]@users.noreply.github.com>

github-actions[bot] committed:
auto updating docs

georgerichardson commented on .github/ISSUE_TEMPLATES/Bug Report.yml:
```diff
@@ -0,0 +1,31 @@
+    id: reproducing
+    attributes:
+      label: Instructions to Reproduce the Bug
+      description: What were the steps you ran that resulted in the bug? Include code snippets if possible.
```
```suggestion
      description: What were the steps you ran that resulted in the bug? Include code snippets or shell commands if possible.
```

georgerichardson commented on .github/ISSUE_TEMPLATES/Feature Request.yml:
```diff
@@ -0,0 +1,28 @@
+name: Feature Request
+description: Request a new feature for nesta_ds_utils
```
```suggestion
description: Request a new feature for `nesta_ds_utils`
```

georgerichardson commented on .github/ISSUE_TEMPLATES/Feature Request.yml:
```diff
@@ -0,0 +1,28 @@
+    id: slack
+    attributes:
+      label: Slack Post
+      description: By submitting this issue, you acknowledge that you have suggested this feature in Slack and it has been approved
```
If using Slack then maybe good to nominate a channel and say which one here. Also not sure who is in charge of approving - maybe 'agreed' is better?

I also wonder if there's a case for putting the feature issue in GitHub first, then telling people on Slack. Keeping a track of the features requested on here is useful for future contributors/requesters, because they can search here for things that have been requested previously and approved or denied.

Don't feel massively strongly about it either way though.

georgerichardson commented on .github/ISSUE_TEMPLATES/Feature Request.yml:
```diff
@@ -0,0 +1,28 @@
+    id: example
+    attributes:
+      label: Examples of this Feature
+      description: If possible, link to a function in an existing Nesta project.
```
```suggestion
      description: If possible, link to a function in an existing project, module, script etc.
```

georgerichardson reviewed (commented):
Great detailed templates!

Have made a couple of minor edits.

Also started a conversation on where the discussion/agreement for new features should happen. Happy to pick this up on Slack if easier. Also fine if you want to make a call on it and we just see how it goes.

emily-bicks committed:
Update .github/ISSUE_TEMPLATES/Bug Report.yml

Co-authored-by: George Richardson <georgerichardson@posteo.net>

emily-bicks committed:
Update .github/ISSUE_TEMPLATES/Feature Request.yml

Co-authored-by: George Richardson <georgerichardson@posteo.net>

emily-bicks committed:
Update .github/ISSUE_TEMPLATES/Feature Request.yml

Co-authored-by: George Richardson <georgerichardson@posteo.net>

Commit:
specifying nesta_ds_utils channel for slack conversation

github-actions[bot] committed:
auto updating docs

Commit:
fixing merge with doctrees environment

github-actions[bot] committed:
auto updating docs

Commit:
only pushing html build files

Commit:
using sphinx-build command rather than make html

github-actions[bot] committed:
auto updating docs

Commit:
Revert "Update .github/ISSUE_TEMPLATES/Feature Request.yml"

This reverts commit d774cf4bede4ae05f77d418e9c60fc530b64dfcf.

reverting to before doc update

Commit:
changing doc build workflow to exclude environment.pickle

Commit:
fixing merge conflict

Commit:
Merge branch 'fixing_merge' into 39_create_contributor_guidelines
fixing merge # the commit.

Commit:
adding pull before commit
